recipes/botan/all/conanfile.py

```diff
@@ -0,0 +1,300 @@
+
+
+class BotanConan(ConanFile):
+    name = 'botan'
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/randombit/botan"
+    license = "BSD-2-Clause"
+    exports = ["patches/*"]
+    description = "Botan is a cryptography library written in C++11."
+    topics = ("cryptography", "crypto", "C++11", "tls")
+    settings = 'os', 'arch', 'compiler', 'build_type'
+    options = {
+        'amalgamation': [True, False],
+        'with_bzip2': [True, False],
+        'with_openssl': [True, False],
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'single_amalgamation': [True, False],
+        'with_sqlite3': [True, False],
+        'with_zlib': [True, False],
+        'with_boost': [True, False],
+        'enable_modules': "ANY",
+        'system_cert_bundle': "ANY"
+    }
+    default_options = {'amalgamation': True,
+                       'with_bzip2': False,
+                       'with_openssl': False,
+                       'shared': True,
+                       'fPIC': True,
+                       'single_amalgamation': False,
+                       'with_sqlite3': False,
+                       'with_zlib': False,
+                       'with_boost': False,
+                       'enable_modules': None,
+                       'system_cert_bundle': None}
+
+    def configure(self):
+        compiler = self.settings.compiler
+        version = Version(self.settings.compiler.version.value)
+
+        if self.settings.os == "Windows" and compiler == "Visual Studio" and version < "14":
+            raise ConanInvalidConfiguration("Botan doesn't support MSVC < 14")
+
+        elif compiler == "gcc" and version >= "5" and compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration(
+                'Using Botan with GCC >= 5 on Linux requires "compiler.libcxx=libstdc++11"')
+
+        elif compiler == "clang" and compiler.libcxx not in ["libstdc++11", "libc++"]:
+            raise ConanInvalidConfiguration(
+                'Using Botan with Clang on Linux requires either "compiler.libcxx=libstdc++11" ' \
+                'or "compiler.libcxx=libc++"')
+
+        if self.options.with_boost:
+            self.options["boost"].add("shared=False")
+            self.options["boost"].add("magic_autolink=False")
+            self.options["boost"].add("without_coroutine=False")
+            self.options["boost"].add("without_system=False")
+
+    def requirements(self):
+        if self.options.with_bzip2:
+            self.requires("bzip2/1.0.6")
+        if self.options.with_openssl:
+            self.requires("openssl/1.0.2t")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_sqlite3:
+            self.requires("sqlite3/3.30.1")
+        if self.options.with_boost:
+            self.requires("boost/1.71.0")
+
+    def config_options(self):
+        if self.options.single_amalgamation:
```

Comment:
move it to `configure(self)`. The method `config_options` is not able to evaluate options, only settings.